### PR TITLE
Merge property file reading logic

### PR DESCRIPTION
**Pull request by Or-Geva**
- [ ] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

### CODE REVIEW

**Review by yahavi — CHANGES_REQUESTED** [review]
I'd like to review the tests after fixing them

File: build-info-extractor/src/test/java/org/jfrog/build/extractor/BuildExtractorUtilsTest.java [hunk 1]
```diff
@@ -115,33 +111,34 @@ public void getBuildInfoPropertiesFromEncryptedFile() throws IOException, Invali
     }
 
     public void failToReadEncryptedFileWithNoKey() throws IOException, InvalidAlgorithmParameterException, NoSuchPaddingException, IllegalBlockSizeException, NoSuchAlgorithmException, BadPaddingException, InvalidKeyException {
-        setupEncryptedFileTest();
+        // Create encrypted file with properties
+        createProperty();
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply build-info-extractor/src/test/java/org/jfrog/build/extractor/BuildExtractorUtilsTest.java:115]: You're right, it should be `setupEncryptedFileTest(createProperty());` and we are removing the keys on line 117

File: build-info-extractor/src/test/java/org/jfrog/build/extractor/BuildExtractorUtilsTest.java [hunk 2]
```diff
@@ -311,4 +325,18 @@ public void testCreateBuildInfoUrl(String url, String buildName, String buildNum
...
     }
+
+    private Properties createProperty() {
+        Properties props = new Properties();
+        props.put(POPO_KEY, "buildname");
+        props.put(MOMO_KEY, "1");
+        return props;
+    }
+
+    private Properties createPropertyEnv() {
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply build-info-extractor/src/test/java/org/jfrog/build/extractor/BuildExtractorUtilsTest.java:336]: It generates slightly different properties apart from using createProperty to specifically handle properties associated with environment reads, distinguishing them from file-related properties.

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**